pkg/config/analysis/analyzers/deployment/services.go

```diff
@@ -152,7 +157,7 @@ func (s *ServiceAssociationAnalyzer) findMatchingServices(r *resource.Instance,
 
 		sSelector := klabels.SelectorFromSet(s.Selector)
 		pLabels := klabels.Set(d.Template.Labels)
-		if sSelector.Matches(pLabels) && r.Metadata.FullName.Namespace.String() == deploymentNS {
+		if !sSelector.Empty() && sSelector.Matches(pLabels) && r.Metadata.FullName.Namespace.String() == deploymentNS {
```

Comment:
@GregHanson Here the logic is to match services and deployments with selectors. However in `default` namespace, there's a service `kubernetes` with no selector. This service will also be matched to deployments, and as a result, the warning message is not generated correctly because `kubernetes` service matches deployments that actually has no service associated in the default namespace.

Reply:
@GregHanson I revised to  check if there are labels existing before matching, which will fix the problem.

Reply:
Open up another PR because it needs releasenotes, and needs to be cherry-picked.

Reply:
sounds good